subsys/bluetooth/shell/bt.c

```diff
@@ -1954,7 +1954,7 @@ static int cmd_adv_data(const struct shell *sh, size_t argc, char *argv[])
 			  appearance);
 
 	ad_len = ad_init(&ad[*data_len], ARRAY_SIZE(ad) - *data_len,
-			 adv_set_opt[selected_adv]);
+			 adv_set_opt[selected_adv]) + *data_len;
```

Comment:
The error below, `shell_error(sh, "Failed to initialize stack advertising data");`, does not match this behavior. 

Suggest to do `ad_len += *data_len` after the `if (ad_len < 0)` check instead